kornia/geometry/liegroup/se2.py

```diff
@@ -0,0 +1,207 @@
+    """
+
+    def __init__(self, r: So2, t: Tensor) -> None:
+        """Constructor for the base class.
+
+        Internally represented by a complex number `z` and a translation 2-vector.
+
+        Args:
+            r: So2 group encompassing a rotation.
+            t: translation vector with the shape of :math:`(B, 2)`.
+
+        Example:
+            >>> so2 = So2.identity()
+            >>> t = torch.ones((1, 2))
+            >>> se2 = Se2(so2, t)
+            >>> se2
+            rotation: (1+0j)
+            translation: Parameter containing:
+            tensor([[1., 1.]], requires_grad=True)
+        """
+        super().__init__()
+        KORNIA_CHECK_TYPE(r, So2)
+        # TODO change to KORNIA_CHECK_SHAPE once there is multiple shape support
+        check_se2_t_shape(t)
+        self._r = r
+        self._t = Parameter(t)
+
+    def __repr__(self) -> str:
+        return f"rotation: {self.r}\ntranslation: {self.t}"
+
+    def __getitem__(self, idx) -> 'Se2':
+        return Se2(self._r[idx], self._t[idx][None])
+
+    def __mul__(self, right: "Se2") -> "Se2":
+        """Compose two Se2 transformations.
+
+        Args:
+            right: the other Se2 transformation.
+
+        Return:
+            The resulting Se2 transformation.
+        """
+        if isinstance(right, Se2):
+            KORNIA_CHECK_TYPE(right, Se2)
+            r = self.so2 * right.so2
+            t = self.t + self.so2 * right.t
+            return Se2(r, t)
+        elif isinstance(right, Tensor):
+            KORNIA_CHECK_TYPE(right, Tensor)
+            # TODO change to KORNIA_CHECK_SHAPE once there is multiple shape support
+            check_se2_t_shape(right)
+            return self.so2 * right + self.t
+        else:
+            raise TypeError(f"Unsupported type: {type(right)}")
+
+    @property
+    def so2(self) -> So2:
+        """Return the underlying rotation(So2)."""
+        return self._r
+
+    @property
+    def r(self) -> So2:
+        """Return the underlying rotation(So2)."""
+        return self._r
+
+    @property
+    def t(self) -> Tensor:
+        """Return the underlying translation vector of shape :math:`(B,3)`."""
+        return self._t
+
+    @staticmethod
+    def exp(v) -> 'Se2':
+        """Converts elements of lie algebra to elements of lie group.
+
+        Args:
+            v: vector of shape :math:`(B, 3)`.
+
+        Example:
+            >>> v = torch.ones((1, 3))
+            >>> s = Se2.exp(v)
+            >>> s.r
+            Parameter containing:
+            tensor([0.5403+0.8415j], requires_grad=True)
+            >>> s.t
+            Parameter containing:
+            tensor([[0.3818, 1.3012]], requires_grad=True)
+        """
+        # TODO when theta is 0
+        check_v_shape(v)
+        theta = v[..., 2]
+        so2 = So2.exp(theta)
+        a = so2.z.imag / theta
+        b = (1.0 - so2.z.real) / theta
+        t = stack((a * v[..., 0] - b * v[..., 1], b * v[..., 0] + a * v[..., 1]), -1)
```

Comment:
```suggestion
        t = stack((a * x - b * y, b * x + a * y), -1)
```
`x, y = v[…, 0], v[…, 1]`